### PR TITLE
Blog app: add devise

### PR DESCRIPTION
**Pull request by ndikumanaisaie**
## In this milestone I covered the following: 
- Remove current_user method in ApplicationController, devise will provide one.
- [Install and setup devise](https://github.com/heartcombo/devise#getting-started).
- Can register a new user.
  - User logs in with a combination of email and password.
  - Hashed password are stored in the database.
  - Ask for confirmation of email.
  - Can reset password.
- [Modified the views in devise](https://github.com/heartcombo/devise#configuring-views) for registration and login to match [the wireframes from the Sneak Peek](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md#:~:text=For%20this%20project%20you%20will%20have%20full%20freedom%20in%20terms%20of%20visual%20design%20but%20you%20will%20need%20to%20keep%20the%20following%20wireframes%3A) and my styling.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,4 +1,8 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (alexander16108) [inline app/models/user.rb]: - [ ] as required your application should be able to ask for confirmation of email. where it's enabled in your user.rb file. you should enable this action in your user.rb file by adding the `: confirmable` keyword.

**Review by alexander16108 — CHANGES_REQUESTED** [review]
Hi @ndikumanaisaie @KIsmoilov , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- Gitflow was correctly used. ✔️
- Good commit messages ✔️
- Professional README ✔️

👍 Good job!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (ndikumanaisaie) [reply app/models/user.rb]: Dear @alexander16108 the confirmable keyword is now added to the user model for an email confirmation.

**Review by KDlamini — APPROVED** [review]
### Hi @ndikumanaisaie 🙋‍♀️,  

Excellent work! 💯 
You have done a great job implementing all the project requirements. 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!🎊 🥳 

## Highlights 🌞 

✔️  User authentication is implemented using Devise
✔️  Email is confirmable
✔️  Good PR title and description

# Status: Approved ✔️ 

giphyapproved
 
Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**